app/policies/proposal_policy/scope.rb

```diff
@@ -0,0 +1,20 @@
+class ProposalPolicy
+  class Scope
+    def initialize(user, scope)
+      @user = user
+      @scope = scope
+    end
+
+    # returns the entire list of Proposals that are visible to the user
+    def resolve
+      if @user.admin?
+        @scope.all
+      elsif @user.client_admin?
+        # TODO include all Proposals that the user is involved in
```

Comment:
Both `Query::` classes are effectively wrappers around existing active record relations (an OO-approach). What do you think about a more declarative style, in which individual clauses can be combined arbitrarily using Arel? This could be implemented, for example, via modules which contained individual Arel "where" clauses (or functions returning such clauses). You could say something like:
```ruby
Proposals.where(Matches::admin(@user).or
                Matches::client_admin(@user).or
                Matches::involved(@user))
```